src/rttable.c

```diff
@@ -216,6 +221,58 @@ void clearAllRoutes(void) {
     my_log(LOG_NOTICE, 0, "All routes removed. Routing table is empty.");
 }
 
+/**
+*	Function which return the number of active dowsnstream hosts
+*/
+int countDownstreamHosts(struct RouteTable *croute) {
+    int result = 0;
+
+    for (int i = 0; i < croute->dwnstrHostsCapacity; i++) {
+        if(croute->dwnstrHosts[i] != 0) {
+            result++;
+        }
+    }
+    return result;
+}
+
+/**
+*	Function to find first downstream host slot
+*/
+size_t findFreeDownstreamSlot(struct RouteTable *croute) {
+    for (size_t i = 0; i < (size_t)croute->dwnstrHostsCapacity; i++) {
+        if(croute->dwnstrHosts[i] == 0) {
+            return i;
+        }
+    }
+    return (size_t)0;
+}
+
+/**
+*	Function to find downstream slot for host
+*/
+size_t findDownstreamSlot(struct RouteTable *croute, uint32_t dstrHost) {
+    for (size_t i = 0; i < (size_t)croute->dwnstrHostsCapacity; i++) {
+        if(croute->dwnstrHosts[i] == dstrHost) {
+            return i;
+        }
+    }
+    return (size_t)-1;
+}
+
+/**
+*	Function to add downstream hosts to route
+*/
+void addDownStreamHost(struct RouteTable *croute, uint32_t dstrHost){
+    croute->dwnstrHostsSize = croute->dwnstrHostsSize + 1;
+
+    if(croute->dwnstrHostsSize > croute->dwnstrHostsCapacity){
+        croute->dwnstrHosts = (uint32_t*)realloc(croute->dwnstrHosts, sizeof(croute->dwnstrHosts) * 2);
```

Comment:
`sizeof(croute->dwnstrHosts)` is always 8 on x86-64.
You probably wanted to use `croute->dwnstrHostsCapacity`.

Reply:
Made a mistake by calling sizeof() on the pointer which indeed is always 8, fixed.